### PR TITLE
Preload the env state in the _app file

### PR DESCRIPTION
**Pull request by Miniplop**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by Miniplop** [comment]
I open this PR to remove all the logic around handling a null env in the frontend part, because the application should not work without it. So this simplify all the selectors in the env module + some component display logic :)

We can use the same logic to hydrate other keys in the store, for exemple the lib/cutom types object when these contexts will be migrated to redux

I follow the best practice written in the redux documentation : https://redux.js.org/usage/structuring-reducers/initializing-state

File: packages/slice-machine/components/App/index.tsx [hunk 1]
```diff
@@ -17,66 +15,43 @@ import useServerState from "@src/hooks/useServerState";
...
-};
-
 type AppProps = {
-  theme: () => Theme;
-  serverState: ServerState | undefined;
+  serverState: ServerState;
 };
 
 const SliceMachineApp: React.FunctionComponent<AppProps> = ({
-  theme,
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/App/index.tsx:37]: Have we gotten rid of theme?

File: packages/slice-machine/src/modules/environment/index.ts [hunk 2]
```diff
@@ -54,22 +45,22 @@ export const getConfigErrors = (store: SliceMachineStoreType): ConfigErrors =>
...
 };
+
 // Reducer
+// This reducer is preloaded by a state coming from the /state call in the _app component
 export const environmentReducer: Reducer<
-  EnvironmentStoreType,
+  EnvironmentStoreType | null,
   EnvironmentActions
-> = (state = initialState, action) => {
+> = (state, action) => {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/src/modules/environment/index.ts:61]: no initialState?

File: packages/slice-machine/pages/_app.tsx [hunk 3]
```diff
@@ -97,15 +122,25 @@ function MyApp({ Component, pageProps }: AppContext & AppInitialProps) {
...
-      <Provider store={store}>
-        <PersistGate loading={null} persistor={persistor}>
-          <TrackerContext.Provider value={tracker}>
-            <SliceMachineApp theme={theme} serverState={serverState}>
-              {isLoadingData ? <LoadingPage /> : <Component {...pageProps} />}
-            </SliceMachineApp>
-          </TrackerContext.Provider>
-        </PersistGate>
-      </Provider>
+      <ThemeProvider theme={theme}>
```
Comment (MarcMcIntosh) [inline packages/slice-machine/pages/_app.tsx:124]: oh, Found the theme

File: packages/slice-machine/pages/_app.tsx [hunk 4]
```diff
@@ -57,13 +71,25 @@ function MyApp({ Component, pageProps }: AppContext & AppInitialProps) {
...
+    store: Store;
+    persistor: Persistor;
+  } | null>(null);
 
   useEffect(() => {
     if (!serverState) {
       return;
     }
+
+    if (!_storeInitiated) {
```
Comment (MarcMcIntosh) [inline packages/slice-machine/pages/_app.tsx]: is `storeInitiated` a state ?

File: packages/slice-machine/pages/_app.tsx [hunk 5]
```diff
@@ -25,6 +26,10 @@ import { LibraryUI } from "lib/models/common/LibraryUI";
 
 import Head from "next/head";
 import { AppInitialProps } from "next/dist/shared/lib/utils";
+import { Store } from "redux";
+import { Persistor } from "redux-persist/es/types";
+
+let _storeInitiated = false;
```
Comment (MarcMcIntosh) [inline packages/slice-machine/pages/_app.tsx]: could this be in the state?

File: packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/index.tsx [hunk 6]
```diff
@@ -95,9 +95,7 @@ const SetupDrawer: React.FunctionComponent<SetupDrawerProps> = ({
             )}
           </Flex>
         </Flex>
-        {!!storybook === false && !!framework && (
-          <StorybookSection framework={framework} />
-        )}
+        {!storybook && <StorybookSection framework={framework} />}
```
Comment (MarcMcIntosh) [inline packages/slice-machine/lib/builders/SliceBuilder/SetupDrawer/index.tsx:98]: ...`!"" &&` yeah that works out :)

File: packages/slice-machine/components/ReviewModal/index.tsx [hunk 7]
```diff
@@ -113,8 +114,7 @@ const ReviewModal: React.FunctionComponent<ReviewModalProps> = () => {
   ): Promise<void> => {
     try {
       startLoadingReview();
-      // env should never be undefined or null
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/ReviewModal/index.tsx:116]: what about the tracker?

File: packages/slice-machine/components/ReviewModal/index.tsx [hunk 8]
```diff
@@ -40,10 +40,11 @@ const ratingSelectable = [1, 2, 3, 4, 5, 6, 7];
 const SelectReviewComponent = ({ field, form }: FieldProps) => {
   return (
     <Box sx={{ mb: 3 }}>
-      {ratingSelectable.map((rating) => (
+      {ratingSelectable.map((index, rating) => (
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/ReviewModal/index.tsx]: pretty sure index is the second parameter for map functions :)

**Review by MarcMcIntosh — CHANGES_REQUESTED** [review]
I've left some questions and observations :)

File: packages/slice-machine/pages/_app.tsx (hunk 3 above)
Comment (Miniplop) [reply packages/slice-machine/pages/_app.tsx:124]: hahaha :)

File: packages/slice-machine/components/App/index.tsx (hunk 1 above)
Comment (Miniplop) [reply packages/slice-machine/components/App/index.tsx:37]: nop just send the them one level above to apply it to the loading page

File: packages/slice-machine/components/ReviewModal/index.tsx (hunk 8 above)
Comment (Miniplop) [reply packages/slice-machine/components/ReviewModal/index.tsx]: ohhhh you right :p

File: packages/slice-machine/src/modules/environment/index.ts (hunk 2 above)
Comment (Miniplop) [reply packages/slice-machine/src/modules/environment/index.ts:61]: Nop as now the default state for the env module is coming from the _app.tsx file :)

File: packages/slice-machine/components/ReviewModal/index.tsx (hunk 7 above)
Comment (Miniplop) [reply packages/slice-machine/components/ReviewModal/index.tsx:116]: not in this PR, but in this PR https://github.com/prismicio/slice-machine/pull/283, in the new tracker implementation I suggest something that get ride of this condition :)

File: packages/slice-machine/components/App/index.tsx (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/components/App/index.tsx:37]: k

File: packages/slice-machine/components/ReviewModal/index.tsx (hunk 7 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/components/ReviewModal/index.tsx:116]: neet

File: packages/slice-machine/components/ReviewModal/index.tsx (hunk 8 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/components/ReviewModal/index.tsx]: you'll get the waring about using the index as the key, so it maybe better to use the rating for the key (assuming the key is unique)

File: packages/slice-machine/pages/_app.tsx (hunk 5 above)
Comment (Miniplop) [reply packages/slice-machine/pages/_app.tsx]: Yes better :)

File: packages/slice-machine/pages/_app.tsx (hunk 4 above)
Comment (Miniplop) [reply packages/slice-machine/pages/_app.tsx]: Fixed :)

File: packages/slice-machine/components/ReviewModal/index.tsx [hunk 9]
```diff
@@ -40,10 +40,11 @@ const ratingSelectable = [1, 2, 3, 4, 5, 6, 7];
 const SelectReviewComponent = ({ field, form }: FieldProps) => {
   return (
     <Box sx={{ mb: 3 }}>
-      {ratingSelectable.map((rating) => (
+      {ratingSelectable.map((rating, index) => (
         <Button
           variant="secondary"
           type="button"
+          key={index}
```
Comment (MarcMcIntosh) [inline packages/slice-machine/components/ReviewModal/index.tsx:47]: this should be a warning.

**Review by MarcMcIntosh — CHANGES_REQUESTED** [review]

**Review by MarcMcIntosh — APPROVED** [review]

**Pull request merged**